### PR TITLE
Add docs directory

### PR DESCRIPTION
**Pull request by man-of-code**

### CODE REVIEW

File: docs/default.md [hunk 1]
```diff
@@ -2,25 +2,11 @@
 
 This is a React + `algob` project which serves as a app template. It has a single package.json file. `algob` works within the project (using `yarn run algob deploy`).
 
-Install `Algo Builder` using
```
Comment (amityadav0) [inline docs/default.md]: Please add link to setup here

File: docs/default.md [hunk 2]
```diff
@@ -2,25 +2,11 @@
...
-- `yarn add @algo-builder/algob` or
-- `yarn link @algo-builder/algob` (with local installation. This is recommended if you want to use `algob` with latest, not released version). Read more about it [here](https://github.com/scale-it/algo-builder#installation).
-
 Before running react app with algosigner:
 - Make sure to add your network config to the algosigner extension (display name, id, algod_url, token). Read [this](https://github.com/PureStake/algosigner#custom-networks) for more information on setting up your wallet in algosigner for custom network.
 - Add your ledger name (display name) in `algorand.config.js` for private-net, otherwise it would be "TestNet" or "MainNet"
 - Custom accounts for private-net in `algob.config.js` can be used within `AlgoSigner.algod` API, but to use them with `AlgoSigner.Accounts`, you need to go to the algosigner extension UI -> import existing account (within your private-net configuration) -> add the account mneumonic (say john's in algob.config.js). Then that account will show in your algosigner wallet as well.
 
-### Adding private-net config to algosigner extension
```
Comment (amityadav0) [inline docs/default.md]: here also, To configure network you can read configure subsection doc.md

File: docs/htlc.md [hunk 3]
```diff
@@ -23,25 +23,21 @@ After that click on the `Withdraw HTLC Escrow`, the webpage will prompt for `rec
 If the receiver's address is not Alice's address, then the code will throw some error. The same applies to Bob's address during the escape condition. 
 Also, the timeout value(3001) of the HTLC should also be kept in mind while executing the template.
 
-
```
Comment (amityadav0) [inline docs/htlc.md]: same

**Comment by man-of-code** [comment]
@amityadav0 All the stup steps have been moved to `setup.md`. That's why I have removed them from template docs.
I'll mention this in the docs too.

File: docs/default.md [hunk 4]
```diff
@@ -2,25 +2,13 @@
...
 
-Install `Algo Builder` using
-- `yarn add @algo-builder/algob` or
-- `yarn link @algo-builder/algob` (with local installation. This is recommended if you want to use `algob` with latest, not released version). Read more about it [here](https://github.com/scale-it/algo-builder#installation).
+NOTE: Please go through `setup.md` first, if not visited.
 
 Before running react app with algosigner:
 - Make sure to add your network config to the algosigner extension (display name, id, algod_url, token). Read [this](https://github.com/PureStake/algosigner#custom-networks) for more information on setting up your wallet in algosigner for custom network.
 - Add your ledger name (display name) in `algorand.config.js` for private-net, otherwise it would be "TestNet" or "MainNet"
 - Custom accounts for private-net in `algob.config.js` can be used within `AlgoSigner.algod` API, but to use them with `AlgoSigner.Accounts`, you need to go to the algosigner extension UI -> import existing account (within your private-net configuration) -> add the account mneumonic (say john's in algob.config.js). Then that account will show in your algosigner wallet as well.
```
Comment (ratik21) [inline docs/default.md]: I think this is also common information accross templates. Can we please move this to the setup/intergrations.md as well?

File: docs/htlc.md [hunk 5]
```diff
@@ -23,25 +25,21 @@ After that click on the `Withdraw HTLC Escrow`, the webpage will prompt for `rec
...
 ## Available Scripts
 
 In the project directory, you can run:
 
 ### Deploy HTLC Escrow Account
 
+Please make sure that you have properly configured the `algob` cli using the steps mentioned in `setup.md`. 
+
+To verify so, execute the following command:
```
Comment (ratik21) [inline docs/htlc.md]:
```suggestion
To verify so, try executing the following command:
```

File: docs/setup.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+```bash
+export PATH="$(yarn global bin):$PATH"
+```
+
+## `AlgoSigner` setup
+
+AlgoSigner provides a very user friendly browser extension. The extension can be easily added by visiting chrome store. To know more about AlgoSigner and to get started with it visit [here](https://github.com/PureStake/algosigner#readme).
+
+After you have succesfully added the browser extension, kindly configure the extension and add your accounts to the respective ledgers.
```
Comment (ratik21) [inline docs/setup.md]:
```suggestion
After successfully adding the browser extension, configure the extension and add your accounts (or import existing ones) to your wallet.
```

File: docs/setup.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+### Adding private-net config to AlgoSigner extension
+
+- Display Name: This is the ledger name of your private-net config. It should be same as `LEDGER` in `/src/algosigner.config.js`.
+- Network ID: This is `<network_name>-<network_id>`. network_name, network_id can be found in `./path_to_node/genesis.json` ("network" & "id" fields).
+- Network Algod URL: This is `http://<network_host>:<network_port>`. network_host, network_port can be found in `./path_to_node/algod.net`.
+- Network Indexer URL: Similar to above, this is URL for indexer.
+- Network Headers: You need to add your `<algod_token>` and `<indexer_token>` here. <indexer_token> could be blank if not needed. Format
+  - ```
+    {"Algod":{"X-Algo-API-Token":"<algod_token>"},"Indexer":{"X-Algo-API-Token":"<indexer_token>"}}
+    ```
```
Comment (ratik21) [inline docs/setup.md]: Please also add a default/example config (user can just copy-paste these value to extension if using same private-net config in `algob.config.js`):
```
Display Name = private-net
Network ID = private-v1
Network Algod URL = http://localhost:4001
Network Indexer URL = // blank
Network Headers = {"Algod":{"X-Algo-API-Token":"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"},"Indexer":{"X-Algo-API-Token":""}}
```

**Review by ratik21 — COMMENTED** [review]
I understand adding `/docs/setup.md`, it is something we discussed. 

But why are we **moving** README files from templates to `/docs`? Let's not do that. Each template should have it's readme. The idea was to move common information from multiple template readme's to `/docs/setup-and-integrations.md`, and add more clear steps there (maybe with more info/links). 
Also we should link each template readme to `/docs`. As when user will use `unbox-template` he will only see the project readme. So it should link to `/docs/setup.md`.

File: docs/default.md (hunk 4 above)
Comment (man-of-code) [reply docs/default.md]: sure.

**Review by ratik21 — APPROVED** [review]

**Pull request merged**